pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_conversion.go

```diff
@@ -0,0 +1,349 @@
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	kube_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+)
+
+func (r *HTTPRouteReconciler) gapiToKumaRule(
+	ctx context.Context, mesh string, route *gatewayapi.HTTPRoute, rule gatewayapi.HTTPRouteRule,
+) (mesh_proto.GatewayRoute_HttpRoute_Rule, *kube_meta.Condition, error) {
+	var backends []*mesh_proto.GatewayRoute_Backend
+
+	for _, backend := range rule.BackendRefs {
+		ref := backend.BackendObjectReference
+
+		destination, condition, err := r.gapiToKumaRef(ctx, mesh, route.Namespace, ref)
+		if err != nil || condition != nil {
+			return mesh_proto.GatewayRoute_HttpRoute_Rule{}, condition, err
+		}
+
+		backends = append(backends, &mesh_proto.GatewayRoute_Backend{
+			// Weight has a default of 1
+			Weight:      uint32(*backend.Weight),
+			Destination: destination,
+		})
+	}
+
+	var matches []*mesh_proto.GatewayRoute_HttpRoute_Match
+
+	for _, match := range rule.Matches {
+		kumaMatch, err := gapiToKumaMatch(match)
+		if err != nil {
+			return mesh_proto.GatewayRoute_HttpRoute_Rule{}, nil, errors.Wrap(err, "couldn't convert match")
+		}
+
+		matches = append(matches, kumaMatch)
+	}
+
+	var filters []*mesh_proto.GatewayRoute_HttpRoute_Filter
+
+	for _, filter := range rule.Filters {
+		kumaFilter, condition, err := r.gapiToKumaFilter(ctx, mesh, route.Namespace, filter)
+		if err != nil || condition != nil {
+			return mesh_proto.GatewayRoute_HttpRoute_Rule{}, condition, err
+		}
+
+		filters = append(filters, kumaFilter)
+	}
+
+	return mesh_proto.GatewayRoute_HttpRoute_Rule{
+		Matches:  matches,
+		Filters:  filters,
+		Backends: backends,
+	}, nil, nil
+}
+
+// gapiToKumaRouteConf converts the route into a route spec and returns any
+// conditions that should be set on parent refs. These are the same across all
+// Kuma parent refs. If a conf cannot be created, it returns a nil conf.
+// It returns error only if an unexpected error has occurred. Issues related to the
+// HTTPRoute spec are reflected in the Conditions.
+func (r *HTTPRouteReconciler) gapiToKumaRouteConf(
+	ctx context.Context, mesh string, route *gatewayapi.HTTPRoute,
+) (*mesh_proto.GatewayRoute_Conf, []kube_meta.Condition, error) {
+	var hostnames []string
+
+	for _, hn := range route.Spec.Hostnames {
+		hostnames = append(hostnames, string(hn))
+	}
+
+	var blockerCondition *kube_meta.Condition
+
+	var rules []*mesh_proto.GatewayRoute_HttpRoute_Rule
+
+	for _, rule := range route.Spec.Rules {
+		kumaRule, condition, err := r.gapiToKumaRule(ctx, mesh, route, rule)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "couldn't convert HTTPRoute to Kuma GatewayRoute")
+		}
+		if condition != nil {
+			blockerCondition = condition
+			break
```

Comment:
Why do we set the variable, break then check it and return? Could just return here if there was a condition set

Reply:
Also OK, just tried to increase the readability inside the for loop and handle returning at the end.